sympy/solvers/solveset.py

```diff
@@ -975,6 +982,438 @@ def _solveset(f, symbol, domain, _check=False):
+        # solving for the sum of more than two powers is possible
+        # but not yet implemented
+        return None
+
+    lhs, rhs = list(ordered(f.args))
+    log_type_equation = expand_log(log(lhs)) - expand_log(log(-rhs))
+
+    return log_type_equation
+
+
+def _is_exponential(f, symbol):
+    r"""
+    Helper to check whether an equation is exponential or not.
+
+    ``f`` is the equation to be checked
+
+    ``symbol`` is the variable in which the equation needs to be checked
+
+    The function extracts each term of the equation and checks if it is
+    of exponential form w.r.t `symbol`.
+    Returns `True` if any term is of exponential type otherwise `False`.
+
+    Examples
+    ========
+
+    >>> from sympy import symbols, cos, exp
+    >>> from sympy.solvers.solveset import _is_exponential as check
+    >>> x, y = symbols('x y')
+    >>> check(x**y - x, y)
+    True
+    >>> check(x**y - x, x)
+    False
+    >>> check(exp(x + 3), x)
+    True
+    >>> check(cos(2**x), x)
+    False
+    """
+
+    expr_args = make_expr_args(f)
+    for expr_arg in expr_args:
+        if isinstance(expr_arg, (Pow, exp)) and (
+                symbol in expr_arg.exp.free_symbols):
+            return True
+    return False
+
+
+def _solve_log(f, symbol):
+    r"""
+    Helper to solve logarithmic equations.
+
+    Logarithmic equation is an equation that involves the logarithm
+    of an expression containing a variable.
+
+    Examples
+    ========
+
+    >>> from sympy import symbols, S, log
+    >>> from sympy.solvers.solveset import _solve_log as solve_log
+    >>> x = symbols('x')
+    >>> f = log(x - 3) + log(x + 3)
+    >>> solve_log(f, x)
+    log((x - 3)*(x + 3))
+
+    * Proof of correctness
+
+    A logarithm is another way to write exponent and is defined by
+
+    .. math:: {log_b x} = y \enspace if \enspace b^y = x
+
+    When one side of the equation contains a single logarithm, the
+    equation can be solved by rewriting the equation as an equivalent
+    exponential equation as defined above. But if one side contains
+    more than one logarithm, we need to use the properties of logarithm
+    to condense it into a single logarithm.
+
+    Take for example
+
+    .. math:: log(2*x) - 15 = 0
+
+    contains single logarithm, therefore we can directly rewrite it to
+    exponential form as
+
+    .. math:: x = exp(15)/2
+
+    But if the equation has more than one logarithm as
+
+    .. math:: log(x - 3) + log(x + 3) = 0
+
+    we use log identities to convert into a reduced form
+
+    Using,
+
+    .. math:: log(a) + log(b) = log(a*b)
+
+    therefore the equation becomes,
+
+    .. math:: log((x - 3)*(x  + 3))
+
+    This equation contains one logarithm and can be solved by rewriting
+    to exponents.
+    """
+
+    lhs, rhs_s = invert_real(f, 0, symbol)
+    rhs = rhs_s.args[0]
+
+    new_lhs = logcombine(lhs, force=True)
+    new_f = new_lhs - rhs
+
+    return new_f
+
+
+def _is_logarithmic(f, symbol):
```

Comment:
Will be adding documentation in the next commit.

Logic has been changed to check each term of the expression of log form rather than previously used logcombine method:

- using logcombine can be risky as it may manipulate the equation like `log(x) - log(2*x)` would reduce to `log(1/2)` and the routine would return `False` as there will be no symbol.